Classes/common/touchdb/TDPuller.m

```diff
@@ -101,8 +104,21 @@ - (void)startChangeTracker
     unsigned heartbeat = $castIf(NSNumber, _options[@"heartbeat"]).unsignedIntValue;
     if (heartbeat >= 15000) _changeTracker.heartbeat = heartbeat / 1000.0;
 
-    NSMutableDictionary* headers = $mdict({ @"User-Agent", [TDRemoteRequest userAgentHeader] });
-    [headers addEntriesFromDictionary:_requestHeaders];
+    //make sure we don't overwrite a custom user-agent header
+    BOOL hasUserAgentHeader = NO;
+    for (NSString *key in self.requestHeaders) {
+        if ([[key lowercaseString] isEqualToString:@"user-agent"]) {
+            hasUserAgentHeader = YES;
+        }
+    }
+    NSMutableDictionary* headers;
+    if (!hasUserAgentHeader) {
+        headers = $mdict({ @"User-Agent", [TDRemoteRequest userAgentHeader] });
+        [headers addEntriesFromDictionary:_requestHeaders];
```

Comment:
I think here we could just `headers[@"User-Agent"] = [TDRemoteRequest userAgentHeader]`? If `-userAgentHeader` could be `nil`, add a check to the `!hasUserAgentHeader`:

```objc
NSString *userAgent = [TDRemoteRequest userAgentHeader];
if (!hasUserAgentHeader && userAgent) {
     headers[@"User-Agent"] = userAgent;
}
```